File: app/src/main/java/com/plweegie/android/squash/modules/SharedPrefModule.java
package com.plweegie.android.squash.modules;


import android.app.Application;
import android.content.SharedPreferences;
import android.preference.PreferenceManager;

import javax.inject.Singleton;

import dagger.Module;
import dagger.Provides;

@Module
public class SharedPrefModule {

    public SharedPrefModule() {}

    @Provides
    @Singleton
    SharedPreferences provideSharedPrefs(Application application) {
        return PreferenceManager.getDefaultSharedPreferences(application);
    }
}

File: app/src/main/java/com/plweegie/android/squash/rest/SharedPrefModule.kt
package com.plweegie.android.squash.rest


import android.app.Application
import android.content.SharedPreferences
import android.preference.PreferenceManager
import dagger.Module
import dagger.Provides
import javax.inject.Singleton

@Module
class SharedPrefModule {

    @Provides
    @Singleton
    fun provideSharedPrefs(application: Application): SharedPreferences =
        PreferenceManager.getDefaultSharedPreferences(application)
}
